### PR TITLE
Add ipfs-cluster builds

### PR DESCRIPTION
**Pull request by hsanjuan**
This:

* Adds ipfs-cluster-ctl and ipfs-cluster-service
* Adds an optional "repo-name" file containing the name of the repository in case it does not match "distname" so that the links to source code and issues work for cluster.
* Rewrites two imports in dist.js without using "destructuring expressions" which make my js interpreter fail.
* Adds a few repo-owner files which were missing, thus fixing links to repositories in a few builds.

link: https://ipfs.io/ipfs/QmQu37NDZcRyWwpxvsRawCRmrAtoLfWMPg71k3gz9rjwU3

### CODE REVIEW

**Comment by hsanjuan** [comment]
@whyrusleeping ok I think this is ready now....

File: dists/ipfs-cluster-service/description [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ipfs-cluster-service runs a full IPFS Cluster peer. These builds are a
+preview and are considered experimental. The last version of ipfs-cluster
```
Comment (whyrusleeping) [inline dists/ipfs-cluster-service/description]: s/last/latest/

File: scripts/dist.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 'use strict'
 
 const fs = require('fs')
-const {join} = require('path')
+const join = require('path').join
```
Comment (whyrusleeping) [inline scripts/dist.js]: did you need to change these for some reason?

File: build-go.sh [hunk 3]
```diff
@@ -99,6 +99,7 @@ function doBuild() {
 	fi
 
 	# now package it all up
+        echo $dir/$binname
```
Comment (whyrusleeping) [inline build-go.sh]: what is this echo for?

File: build-go.sh [hunk 4]
```diff
@@ -138,7 +139,12 @@ function printInitialDistfile() {
 	local version=$2
 	test -e description || fail "no description file found"
 
-	printf "{\"id\":\"$distname\",\"version\":\"$version\",\"releaseLink\":\"/$distname/$version\"}" |
+	local reponame=$distname
```
Comment (whyrusleeping) [inline build-go.sh]: quotes around $distname

File: build-go.sh [hunk 5]
```diff
@@ -138,7 +139,12 @@ function printInitialDistfile() {
...
+	if [ -e repo-name ]; then
+		reponame=`cat repo-name`
```
Comment (whyrusleeping) [inline build-go.sh]: I prefer using `$(thing)`

File: scripts/dist.js (hunk 2 above)
Comment (whyrusleeping) [inline scripts/dist.js]: nvm, read the top comment

File: build-go.sh (hunk 3 above)
Comment (hsanjuan) [inline build-go.sh]: good catch, leftover

**Comment by hsanjuan** [comment]
Thanks @whyrusleeping. Addressed all comments.

**Pull request merged**